Mage/src/main/java/mage/watchers/common/FirstStrikeDamageWatcher.java

```diff
@@ -0,0 +1,57 @@
+package mage.watchers.common;
+
+import mage.MageObjectReference;
+import mage.constants.WatcherScope;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.watchers.Watcher;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.UUID;
+
+/**
+ * @author xenohedron
+ */
+public class FirstStrikeDamageWatcher extends Watcher {
```

Comment:
name of the Watcher, name of the dealtFirstStrikeDamage, and comment on the Set could be rewritten with Damage => Step, now that damage is no longer involved.